### PR TITLE
Integrate the correlation algorithm into an express api

### PR DESCRIPTION
**Pull request by Lev-Stambler**
Integrate the correlation search into an express api

Impacted is used as the search query. I was hoping to make an overarching, refactor PR or another forked repo entirely which is more focused on the search aspect

[The Schopal Site](https://schopal.neocities.org)

### CODE REVIEW

File: apps/api-express/src/app/controllers/search.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { findQueryResults } from '../daos';
+
+const router = express.Router();
+
+/**
+ * Search the database for correlated paragraphs
+ * @query q - query string
+ * @query maxNumberOfParagraphs - optional max number of paragraphs
+ * @query numberOfArticles - optional number of articles to search
+ */
```
Comment (cwlarson-google) [inline apps/api-express/src/app/controllers/search.ts:11]: These params are wrong

File: apps/api-express/src/app/daos/search.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import {
+  ParsedArticleParagraphStandalone,
+  ParsedArticle,
+  ParsedArticleParagraph,
+  ArticleParagraphBacksUpClaim,
+} from '@foodmedicine/interfaces';
+import * as scholarsScraper from '@foodmedicine/scholars-scraper';
```
Comment (cwlarson-google) [inline apps/api-express/src/app/daos/search.ts:7]: Why not `import { runScholarScraper } from '@foodmedicine/scholars-scraper';`?

File: apps/api-express/src/app/controllers/search.ts (hunk 1 above)
Comment (Lev-Stambler) [reply apps/api-express/src/app/controllers/search.ts:11]: How so?

File: apps/api-express/src/app/daos/search.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import * as articleParser from '@foodmedicine/article-parser';
+
+export async function findQueryResults(
+  query: string,
+  opts?: {
+    numberOfArticles?: number;
+    maxNumberOfParagraphs?: number;
+  }
+): Promise<ParsedArticleParagraphStandalone[]> {
+  // The query is the impact and the recommendation is left blank
```
Comment (vitashubin) [inline apps/api-express/src/app/daos/search.ts:17]: Can you clarify this comment?  Not sure I understand it

File: apps/api-express/src/app/daos/search.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  const articleHeads = await runScholarsScraper(
+    query,
+    // TODO this is a temporary fix, removing the entire recommendation, impact abstraction should be done
```
Comment (vitashubin) [inline apps/api-express/src/app/daos/search.ts:20]: Do you mean passing in query twice is a temp fix? Could you instead pass in an empty string?

File: apps/api-express/src/app/daos/search.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      (paragraph: ParsedArticleParagraph) => {
+        return {
+          head: article.head,
+          // set default backsUpClaim to notApplicable. This later gets changed manually in the JSON file
+          backsUpClaim: ArticleParagraphBacksUpClaim.notApplicable,
+          ...paragraph,
+        };
+      }
+    );
+    allParagraphsStandalone.push(...standaloneParagraphs);
```
Comment (vitashubin) [inline apps/api-express/src/app/daos/search.ts:48]: I think the forEach, nested .map() and pushing a ...[] can be a bit confusing. If there's no simpler implementation, can you leave a comment explaining what the expected outcome is to help readers understand what this should look like?

File: apps/table-frontend-e2e/src/support/commands.ts [hunk 6]
```diff
@@ -16,7 +16,7 @@ declare namespace Cypress {
 //
 // -- This is a parent command --
 Cypress.Commands.add('login', (email, password) => {
-  console.log('Custom command example: Login', email, password);
+  console.info('Custom command example: Login', email, password);
```
Comment (vitashubin) [inline apps/table-frontend-e2e/src/support/commands.ts:19]: I'm guessing you don't want to be logging passwords here?

File: apps/api-express/src/app/controllers/search.ts (hunk 1 above)
Comment (cwlarson-google) [reply apps/api-express/src/app/controllers/search.ts:11]: oh my bad, I thought these were `@param`'s

File: apps/table-frontend-e2e/src/support/commands.ts (hunk 6 above)
Comment (Lev-Stambler) [reply apps/table-frontend-e2e/src/support/commands.ts:19]: Oh yeah, I did not even check the e2e directory as it was auto generated and I do not have e2e tests yet, but will remove!

File: apps/api-express/src/app/daos/search.ts (hunk 5 above)
Comment (Lev-Stambler) [reply apps/api-express/src/app/daos/search.ts:48]: Ok sure, added

File: apps/api-express/src/app/daos/search.ts (hunk 4 above)
Comment (Lev-Stambler) [reply apps/api-express/src/app/daos/search.ts:20]: Oh yeah, it passed in twice in order to not interfere with the cross feature

File: apps/api-express/src/app/daos/search.ts (hunk 3 above)
Comment (Lev-Stambler) [reply apps/api-express/src/app/daos/search.ts:17]: Oh the comment is now changed to match with passing query in twice

File: apps/api-frontend/src/app/pages/search-result.tsx [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+  paragraph: ParsedArticleParagraphStandalone;
+  key: string;
+}) {
+  return (
+    <div
+      className="single-result-container"
+      onClick={() => window.open(props.paragraph.head.xmlFullTextDownloadLink)}
+    >
+      <h4 aria-label="paper's title">{props.paragraph.head.title}</h4>
+      <p aria-label="correlated paragraph">...{props.paragraph.body}...</p>
```
Comment (vitashubin) [inline apps/api-frontend/src/app/pages/search-result.tsx:20]: You can remove both of these aria labels. You usually should not have an aria label on text elements :)

File: apps/api-frontend/src/index.html [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <title>Schopal</title>
+    <base href="/" />
+
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link rel="icon" type="image/x-icon" href="favicon.ico" />
```
Comment (vitashubin) [inline apps/api-frontend/src/index.html:9]: Looks like this is an icon-based link? If so, this would be a good place to have an aria label

**Pull request merged**